platform/mbed_wait_api_rtos.cpp

```diff
@@ -25,11 +25,11 @@
 #include "platform/mbed_power_mgmt.h"
 
 void wait(float s) {
-    wait_us(s * 1000000.0f);
+    wait_ms(s * 1000.0f);
 }
 
 void wait_ms(int ms) {
-    wait_us(ms * 1000);
+    Thread::wait((uint32_t)ms);
```

Comment:
Looking at the old function there are some differences. That is if you enter a delay that has some us component. The old function would do a thread wait and then a busy wait:
wait_us(1500) would call  Thread::wait(1); and then do a busy wait for the remaining 500 us

In this sense wait_s(float s) is quite "non deterministic". A slight change in the input to wait_s() (eg 0.002 vs 0.001999) can have a big input to the power consumption. This might also be the issue that @stevew817 was actually trying to address.